### PR TITLE
Add SON's mechanism into nodes

### PR DESCRIPTION
**Pull request by pixelplex**
This PR includes general mechanisms for SON user role. Includes:
- Voting and electing active SONs among all SONs
- Vesting requirement to join SONs
- Introduced operations and objects as per the suggested objects and operations list
- Fixed build errors

### CODE REVIEW

File: libraries/app/include/graphene/app/database_api.hpp [hunk 1]
```diff
@@ -44,6 +44,8 @@
 #include <graphene/chain/betting_market_object.hpp>
 #include <graphene/chain/global_betting_statistics_object.hpp>
 
+#include <graphene/chain/btc-sidechain/son.hpp>
```
Comment (bobinson) [inline libraries/app/include/graphene/app/database_api.hpp]: @pixelplex  - provide the LLD where this implementation is designed.

File: Dockerfile [hunk 2]
```diff
@@ -36,10 +38,18 @@ RUN \
 	done && \
       git submodule sync --recursive ) && \
     git submodule update --init --recursive && \
+    BOOST_ROOT=$HOME/opt/boost_1_60_0 && \
+    wget -c 'http://sourceforge.net/projects/boost/files/boost/1.60.0/boost_1_60_0.tar.gz/download' -O boost_1_60_0.tar.gz &&\
```
Comment (bobinson) [inline Dockerfile]: Wrong Boost version is used.

File: libraries/chain/btc-sidechain/son_operations_evaluator.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#include <graphene/chain/btc-sidechain/son_operations_evaluator.hpp>
+
+#include <graphene/chain/database.hpp>
+#include <graphene/chain/btc-sidechain/son.hpp>
```
Comment (bobinson) [inline libraries/chain/btc-sidechain/son_operations_evaluator.cpp]: Please provide the LLD where the changes done here are designed.

File: libraries/chain/db_block.cpp [hunk 4]
```diff
@@ -43,6 +43,8 @@
 
 #include <fc/smart_ref_impl.hpp>
 
+#include <graphene/chain/btc-sidechain/son.hpp>
```
Comment (bobinson) [inline libraries/chain/db_block.cpp]: Provide LLD

File: libraries/chain/db_init.cpp [hunk 5]
```diff
@@ -77,8 +77,12 @@
 #include <graphene/chain/betting_market_evaluator.hpp>
 #include <graphene/chain/tournament_evaluator.hpp>
 
+#include <graphene/chain/btc-sidechain/son_operations_evaluator.hpp>
```
Comment (bobinson) [inline libraries/chain/db_init.cpp]: provide LLD

File: libraries/chain/include/graphene/chain/btc-sidechain/son_operations_evaluator.hpp [hunk 6]
```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include <graphene/chain/evaluator.hpp>
+#include <graphene/chain/btc-sidechain/son_operations.hpp>
```
Comment (bobinson) [inline libraries/chain/include/graphene/chain/btc-sidechain/son_operations_evaluator.hpp]: provide the LLD with the UML

File: libraries/chain/son_operations_evaluator.cpp [hunk 7]
```diff
@@ -0,0 +1,44 @@
+#include <graphene/chain/btc-sidechain/son_operations_evaluator.hpp>
+
+#include <graphene/chain/database.hpp>
+#include <graphene/chain/btc-sidechain/son.hpp>
```
Comment (bobinson) [inline libraries/chain/son_operations_evaluator.cpp]: Provide the LLD where these changes are captured with UMLs

File: tests/common/database_fixture.cpp [hunk 8]
```diff
@@ -46,6 +46,8 @@
 #include <graphene/chain/event_object.hpp>
 #include <graphene/chain/tournament_object.hpp>
 
+#include <graphene/chain/btc-sidechain/son.hpp>
```
Comment (bobinson) [inline tests/common/database_fixture.cpp]: Provide the LLD where these changes are captured with UMLs

**Comment by bobinson** [comment]
The code needs to refactored to meet the comments made on the LLDs shared.

**Comment by pixelplex** [comment]
will be divided into several requests

**Pull request closed without merging**